src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -306,6 +314,196 @@ void DatasetManager::HandleGet(Coap::Header &aHeader, Message &aMessage, const I
+    {
+        tlv.SetType(OT_MESHCOP_TLV_PENDINGTIMESTAMP);
+        tlv.SetLength(sizeof(aDataset.mPendingTimestamp));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        data64 = HostSwap64(aDataset.mPendingTimestamp);
+        SuccessOrExit(error = message->Append(&data64, sizeof(aDataset.mPendingTimestamp)));
+    }
+
+    if (aDataset.mIsMasterKeySet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_MASTERKEY);
+        tlv.SetLength(sizeof(aDataset.mMasterKey));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        SuccessOrExit(error = message->Append(aDataset.mMasterKey.m8, sizeof(aDataset.mMasterKey.m8)));
+    }
+
+    if (aDataset.mIsNetworkNameSet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_NETWORKNAME);
+        tlv.SetLength(static_cast<uint8_t>(strlen(aDataset.mNetworkName.m8)));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        SuccessOrExit(error = message->Append(aDataset.mNetworkName.m8,
+                                              static_cast<uint8_t>(strlen(aDataset.mNetworkName.m8))));
+    }
+
+    if (aDataset.mIsExtendedPanIdSet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_EXTPANID);
+        tlv.SetLength(sizeof(aDataset.mExtendedPanId));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        SuccessOrExit(error = message->Append(aDataset.mExtendedPanId.m8, sizeof(aDataset.mExtendedPanId.m8)));
+    }
+
+    if (aDataset.mIsMeshLocalPrefixSet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_LOCALPREFIX);
+        tlv.SetLength(sizeof(aDataset.mMeshLocalPrefix));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        SuccessOrExit(error = message->Append(aDataset.mMeshLocalPrefix.m8, sizeof(aDataset.mMeshLocalPrefix.m8)));
+    }
+
+    if (aDataset.mIsDelaySet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_DELAYTIMER);
+        tlv.SetLength(sizeof(aDataset.mDelay));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        data32 = HostSwap32(aDataset.mDelay);
+        SuccessOrExit(error = message->Append(&data32, sizeof(aDataset.mDelay)));
+    }
+
+    if (aDataset.mIsPanIdSet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_PANID);
+        tlv.SetLength(sizeof(aDataset.mPanId));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        data16 = HostSwap16(static_cast<uint16_t>(aDataset.mPanId));
+        SuccessOrExit(error = message->Append(&data16, sizeof(aDataset.mPanId)));
+    }
+
+    if (aDataset.mIsChannelSet)
+    {
+        tlv.SetType(OT_MESHCOP_TLV_CHANNEL);
+        tlv.SetLength(sizeof(aDataset.mChannel));
+        SuccessOrExit(error = message->Append(&tlv, sizeof(tlv)));
+        data16 = HostSwap16(aDataset.mChannel);
+        SuccessOrExit(error = message->Append(&data16, sizeof(aDataset.mChannel)));
+    }
+
+    if (aSize > 0)
+    {
+        SuccessOrExit(error = message->Append(aTlvs, aSize));
+    }
+
+    mMle.GetLeaderAddress(leader);
```

Comment:
You can write the address directly into `messageInfo` and remove the need for having `leader`.